internal/factsengine/policy.go

```diff
@@ -0,0 +1,123 @@
+package factsengine
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/internal/factsengine/gatherers"
+)
+
+const (
+	factsGatheringRequestEvent = "trento.checks.v1.FactsRequest"
+	factsGatheredEvent         = "trento.checks.v1.FactsGathered"
+	eventSource                = "https://github.com/trento-project/agent/internal/factsengine"
+)
+
+func (c *FactsEngine) handleEvent(contentType string, request []byte) error {
+	event, err := handleContentType(contentType, request)
+	if err != nil {
+		return errors.Wrap(err, "Error handling event")
+	}
+
+	switch eventType := event.Context.GetType(); eventType {
+	case factsGatheringRequestEvent:
+		err := c.handleFactsGatheringRequestEvent(event.DataEncoded)
+		if err != nil {
+			return errors.Wrap(err, "Error handling facts request")
+		}
+	default:
+		return fmt.Errorf("Invalid event type: %s", eventType)
+	}
+	return nil
+}
+
+func handleContentType(contentType string, event []byte) (*cloudevents.Event, error) {
+	switch contentType {
+	case cloudevents.ApplicationCloudEventsJSON:
```

Comment:
Interesting, this would allow the agent to ingest different types of integration events next to the cloudevents.

Let's keep an eye on this and see how far to go.